src/utils/transition-manager.js

```diff
@@ -175,15 +195,16 @@ export default class TransitionManager {
     const {startTime, duration, easing, interpolator, startProps, endProps} = this.state;
 
     let shouldEnd = false;
-    let t = (currentTime - startTime) / duration;
+    let t = (currentTime - startTime) / duration + this.state.completion;
```

Comment:
This is incorrect. When we update the transition from A -> B to A -> C, what should happen at the break point is:

![image](https://user-images.githubusercontent.com/2059298/47319126-bdf48f00-d602-11e8-8648-032323ffd729.png)

Your current solution jumps:
![image](https://user-images.githubusercontent.com/2059298/47319198-e1b7d500-d602-11e8-8f14-1a2e76faf644.png)